python/graphstorm/config/argument.py

```diff
@@ -816,16 +817,22 @@ def restore_model_layers(self):
         """ GraphStorm model layers to load.
         """
         # pylint: disable=no-member
+        model_layers = GRAPHSTORM_MODEL_ALL_LAYERS
         if hasattr(self, "_restore_model_layers"):
             assert self.restore_model_path is not None, \
                 "restore-model-path must be provided if restore-model-layers is specified."
             model_layers = self._restore_model_layers.split(',')
             for layer in model_layers:
                 assert layer in GRAPHSTORM_MODEL_LAYER_OPTIONS, \
                     f"{layer} is not supported, must be any of {GRAPHSTORM_MODEL_LAYER_OPTIONS}"
-            return model_layers
-
-        return GRAPHSTORM_MODEL_ALL_LAYERS
+        if self.training_method["name"] == "glem":
+            # GLEM restore model layers conflicts with all layers
+            if model_layers == GRAPHSTORM_MODEL_ALL_LAYERS:
+                logging.warning("Restoring GLEM's LM from checkpoint doesn't support" \
+                                + " all layers, setting to: '%s'", GRAPHSTORM_MODEL_EMBED_LAYER
+                                )
+                model_layers = [GRAPHSTORM_MODEL_EMBED_LAYER]
```

Comment:
i'm not sure why we need this warning.

Reply:
for GLEM, do we need users to specify `restore_model_layers `?

Reply:
Yes user will need to specify `restore_model_layers`. This is for loading a pre-trained LM, possible options are `['embed']` and `['embed', 'decoder']`. If not specified, we will default to `['embed']` and print this warning.

Reply:
Then the warning message should print the possible value of `restore_model_layers`